Add DB limitation to data sandboxes (#11921)
metabase/metabase: docs/enterprise-guide/data-sandboxes.md

I.e., if a user belongs to two user groups, both of which have been given sandboxed access to the same table, that user will not be able to access data from that table. You will either need to remove that user from one of those groups, or remove the sandboxed access from one of those groups.

#### Data sandboxes do not support non-SQL databases

> Non-SQL databases such as Google Analytics, Apache Druid, and MongoDB are not currently supported by data sandboxes.

Reviewer: ```suggestion
You can't currently create data sandboxes with non-SQL databases such as Google Analytics, Apache Druid, or MongoDB .
```
